python/ray/serve/storage/kv_store.py

```diff
@@ -57,13 +60,16 @@ def put(self, key: str, val: bytes) -> bool:
         if not isinstance(val, bytes):
             raise TypeError("val must be bytes, got: {}.".format(type(val)))
 
-        return self.gcs_client.internal_kv_put(
-            self.get_storage_key(key).encode(),
-            val,
-            overwrite=True,
-            namespace=ray_constants.KV_NAMESPACE_SERVE,
-            timeout=self.timeout,
-        )
+        try:
+            return self.gcs_client.internal_kv_put(
+                self.get_storage_key(key).encode(),
+                val,
+                overwrite=True,
+                namespace=ray_constants.KV_NAMESPACE_SERVE,
+                timeout=self.timeout,
+            )
+        except Exception as e:
```

Comment:
Is there a reason the `gcs_client` can't directly raise a `KVStoreError`? What is the catch-call behavior for?

Reply:
gcs client is lower layer things and just a gRPC client wrapper. How to handle the error should be the application code.

The catch and re-raise is to convert the gcs client error to the one serve recognize.

I think for S3 or other storage, it should do similar things.